main.py

```diff
@@ -117,60 +150,85 @@ def generate_sample(dataset, sample_dir, samples = None):
+                                                            duplicates_file = os.path.join(dedup_dir, dataset.name + '_Non-Unique_Texts.csv'))
+                else:
+                    dataset_index_max = Deduplicator.get_length(dataset)
 
+                # Get Quality dictionary
                 if get_quality or manifest.get('stats',{}).get('quality',None):
                     quality_count = {'LOW': 0, 'MEDIUM': 0, 'HIGH': 0}
- 
-                ar = Archive(os.path.join(base_dir, "data"))
-                with Pool(initializer=initialize_worker, processes=args.processes,
-                          maxtasksperchild=maxtasksperchild) as pool:
-                    for txt, meta, index in pool.imap(func=process_doc_partial, iterable=enumerate(ds),
-                                                         chunksize=1):
-                        
+
+                # Init Archive for final dataset file
+                ar = Archive(os.path.join(base_dir, TEMP_DATA))
+
+                ds_extdata = dataset.ext_data
+
+                with Pool(initializer = initialize_worker, processes = args.processes,
+                          maxtasksperchild = maxtasksperchild) as pool:
+
+                    for txt, meta, index in tqdm(pool.imap(func = process_doc_partial,
+                                                      iterable = enumerate(ds_extdata),
+                                                      chunksize = 1),
+                                                total = dataset_index_max,
+                                                smoothing=0.01):
+
+                        name = meta.get("name", meta.get("url", ""))
+
+                        # Check if document is a duplicate
                         if get_duplicates and index in duplicate_indices:
-                            name = meta.get("name", "") or meta.get("url", "")[:80]
-                            log("Removed duplicate : " + name, "WARNING")
+                            logging.warning(f"Removed duplicate : {name}")
                             continue
 
-                        if txt and len(txt) > 200 and meta['words'] > 0:
-                                                        
-                            if get_lang and not meta['language']['lang'] == 'pl':
-                                try:
-                                    name = meta.get("name", "") or meta.get("url", "")[:80]
-                                    log("Removed non 'pl' document : " + name, "WARNING")
-                                except:
-                                    pass
+                        # Check if document has minimum length and words
+                        if txt and len(txt) > MIN_TXT_LENGTH and meta['words'] > 0:
+
+                            # Check for document language
+                            if get_lang and meta['language']['lang'] != 'pl':
```

Comment:
If we are comparing strings, it is safe to compare the same letters case like lower or upper:
`if get_lang and meta['language']['lang'].lower() != 'pl':`